client/sniffer/src/main/java/org/opensearch/client/sniff/OpenSearchNodesSniffer.java

```diff
@@ -241,74 +242,23 @@ private static Node readNode(String nodeId, JsonParser parser, Scheme scheme) th
         }
 
         Map<String, List<String>> realAttributes = new HashMap<>(protoAttributes.size());
-        List<String> keys = new ArrayList<>(protoAttributes.keySet());
-        for (String key : keys) {
-            if (key.endsWith(".0")) {
-                String realKey = key.substring(0, key.length() - 2);
-                List<String> values = new ArrayList<>();
-                int i = 0;
-                while (true) {
-                    String value = protoAttributes.remove(realKey + "." + i);
-                    if (value == null) {
-                        break;
-                    }
-                    values.add(value);
-                    i++;
-                }
-                realAttributes.put(realKey, unmodifiableList(values));
-            }
-        }
         for (Map.Entry<String, String> entry : protoAttributes.entrySet()) {
-            realAttributes.put(entry.getKey(), singletonList(entry.getValue()));
-        }
-
-        if (version.startsWith("2.")) {
-            /*
-             * 2.x doesn't send roles, instead we try to read them from
-             * attributes.
-             */
-            boolean clientAttribute = v2RoleAttributeValue(realAttributes, "client", false);
-            Boolean masterAttribute = v2RoleAttributeValue(realAttributes, "master", null);
-            Boolean dataAttribute = v2RoleAttributeValue(realAttributes, "data", null);
-            if ((masterAttribute == null && false == clientAttribute) || masterAttribute) {
-                roles.add("master");
+            if (entry.getValue().startsWith("[")) {
```

Comment:
I certainly like this encoding approach better, but won't that be a breaking change in a sense that whoever used old notation would suddenly run into issues?

Reply:
The array notation for the setting value in Rest API Response has been changed since Elasticsearch 6.1, and Elasticsearch users are supposed to at least upgrade node to version 6.8 before rolling upgrade to OpenSearch, so I don't think this code worth to be compatible with ES version <= 6.0 .
In ES version >= 6.1, the array notation in API response has been `"attributes": { "array": "[m, 1]" }`.
I will ask for opinion with others as well. 😁

Reply:
Probably to avoid breaking change, I could add the old codes back to parse the old array notation `"attributes": { "array.0": "m", "array.1": "1" }` when backing porting this code change to 2.x branch.

Reply:
> so I don't think this code worth to be compatible with ES version <= 6.0 .
💯 agree to that, thanks @tlfeng